third_party/eigen_fix_gpu_compilation.patch

```diff
@@ -0,0 +1,19 @@
+diff -Naur eigen-eigen-8475e3c056d3/Eigen/src/Core/arch/GPU/PacketMathHalf.h eigen_archive/Eigen/src/Core/arch/GPU/PacketMathHalf.h
+--- eigen-eigen-8475e3c056d3/Eigen/src/Core/arch/GPU/PacketMathHalf.h	     2018-07-12 04:07:16.000000000 -0400
++++ eigen_archive/Eigen/src/Core/arch/GPU/PacketMathHalf.h		     2018-07-18 15:14:40.001758133 -0400
+@@ -945,10 +945,10 @@
+     AlignedOnScalar = 1,
+     size = 8,
+     HasHalfPacket = 0,
+-    HasAdd    = 1,
```

Comment:
Wouldn't this would cause a significant performance regression for GPUs that support these packet ops? Can you think of a less invasive workaround?

Reply:
The true fix for this workaround is here:

https://bitbucket.org/eigen/eigen/commits/43b0af1f9a03d38d5296bda378c299e80a7075b1

It happens to be in a commit that is a few commits ahead of the one that is being discussed here. You can cherry-pick/merge that one directly (I believe we only need the one line change in 
` unsupported/Eigen/CXX11/src/Tensor/TensorFunctors.h `) and that ought to remove the need for the patch file.